File: tests/ruby/src/T014_MessageSpecification.rb
#!/usr/bin/env ruby

require 'libgmsec_ruby'

load 'Test.rb'


class Test_MessageSpecification < Test

	def initialize(args)
		super(args)
	end


    def test()
        test_message_specification
	end


    def test_message_specification()
        Libgmsec_ruby::Log::info("Test MessageSpecification")

        test_message_specification_aux(Libgmsec_ruby::GMSEC_MSG_SPEC_CURRENT, Libgmsec_ruby::Specification::SchemaLevel_LEVEL_0)
        test_message_specification_aux(Libgmsec_ruby::GMSEC_MSG_SPEC_CURRENT, Libgmsec_ruby::Specification::SchemaLevel_LEVEL_1)
        test_message_specification_aux(Libgmsec_ruby::GMSEC_MSG_SPEC_CURRENT, Libgmsec_ruby::Specification::SchemaLevel_LEVEL_2)

        test_message_specification_aux(Libgmsec_ruby::GMSEC_MSG_SPEC_2019_00, Libgmsec_ruby::Specification::SchemaLevel_LEVEL_0)
        test_message_specification_aux(Libgmsec_ruby::GMSEC_MSG_SPEC_2019_00, Libgmsec_ruby::Specification::SchemaLevel_LEVEL_1)
        test_message_specification_aux(Libgmsec_ruby::GMSEC_MSG_SPEC_2019_00, Libgmsec_ruby::Specification::SchemaLevel_LEVEL_2)
	end


    def test_message_specification_aux(specVersion, schemaLevel)
        config = Libgmsec_ruby::Config.new()

        config.add_value("gmsec-specification-version", "" + specVersion.to_s)
        config.add_value("gmsec-schema-level", "" + schemaLevel.to_s)

        spec = Libgmsec_ruby::Specification.new(config)

        msgSpecs = spec.get_message_specifications()

        check("Expected at least 1 (one) message specification", msgSpecs.size() > 0)

        for msgSpec in msgSpecs do
            check("Schema ID is NULL", msgSpec.get_schema_id() != nil)
            check("Subject Template is NULL", msgSpec.get_subject_template() != nil)

            fieldSpecs = msgSpec.get_field_specifications()

            check("Expected at least 1 (one) field specification", fieldSpecs.size() > 0)
		end
	end
end


begin
    test = Test_MessageSpecification.new( ARGV )
    test.run
end

File: tests/python3/src/T014_MessageSpecification.py
#!/usr/bin/env python3

import libgmsec_python3 as lp
from Test import Test


class Test_MessageSpecification(Test):

    def test(self):
        self.test_message_specification()


    def test_message_specification(self):
        lp.log_info("Test MessageSpecification")

        self.test_message_specification_aux(lp.GMSEC_MSG_SPEC_CURRENT, lp.Specification.SchemaLevel_LEVEL_0)
        self.test_message_specification_aux(lp.GMSEC_MSG_SPEC_CURRENT, lp.Specification.SchemaLevel_LEVEL_1)
        self.test_message_specification_aux(lp.GMSEC_MSG_SPEC_CURRENT, lp.Specification.SchemaLevel_LEVEL_2)

        self.test_message_specification_aux(lp.GMSEC_MSG_SPEC_2019_00, lp.Specification.SchemaLevel_LEVEL_0)
        self.test_message_specification_aux(lp.GMSEC_MSG_SPEC_2019_00, lp.Specification.SchemaLevel_LEVEL_1)
        self.test_message_specification_aux(lp.GMSEC_MSG_SPEC_2019_00, lp.Specification.SchemaLevel_LEVEL_2)


    def test_message_specification_aux(self, specVersion, schemaLevel):
        config = lp.Config()

        config.add_value("gmsec-specification-version", "" + str(specVersion))
        config.add_value("gmsec-schema-level", "" + str(schemaLevel))

        spec = lp.Specification(config)

        msgSpecs = spec.get_message_specifications()

        self.check("Expected at least 1 (one) message specification", msgSpecs.size() > 0)

        for msgSpec in msgSpecs:
            self.check("Schema ID is NULL", msgSpec.get_schema_id() != None)
            self.check("Subject Template is NULL", msgSpec.get_subject_template() != None)

            fieldSpecs = msgSpec.get_field_specifications()

            self.check("Expected at least 1 (one) field specification", fieldSpecs.size() > 0)


Test.run('Test MessageSpecification', Test_MessageSpecification())
